### PR TITLE
Add `High Quality` sun sampling mode. Fix issue where enabling draw sun but not enabling sunlight would draw a sun with no contribution to the scene.

### LINKED ISSUE #1184: Add a way to disable NEE for the sun.
While sunlit scenes without sun NEE this would take much longer to converge, would this fix the absence of caustics in non-still water?
It would also be a nice feature for those who want to experiment with Chunky and path tracing.

### PR DESCRIPTION
**Pull request by ThatRedox**
Closes #1184.
This adds a `Sun Luminosity` setting which changes the true brightness of the sun when intersected with organically and a new sun sampling mode which does both sun samples and allows for sun intersections for diffuse materials. This new sun sampling mode allows for visual effects like caustics (with much render time):
image

### CODE REVIEW

**Comment by leMaik** [comment]
Nice! :)

> Fix issue where enabling draw sun but not enabling sunlight would draw a sun with no contribution to the scene.

I always assumed this would be a feature so that one can use sky light and emitters only but still have a visual sun?

File: chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java [hunk 1]
```diff
@@ -2896,7 +2897,13 @@ public synchronized void importFromJson(JsonObject json) {
     saveSnapshots = json.get("saveSnapshots").boolValue(saveSnapshots);
     emittersEnabled = json.get("emittersEnabled").boolValue(emittersEnabled);
     emitterIntensity = json.get("emitterIntensity").doubleValue(emitterIntensity);
-    sunEnabled = json.get("sunEnabled").boolValue(sunEnabled);
+
+    if (json.get("sunEnabled").boolValue(false)) {
+      sunSamplingStrategy = SunSamplingStrategy.Fast;
+    } else {
+      sunSamplingStrategy = SunSamplingStrategy.valueOf(json.get("sunSamplingStrategy").asString(SunSamplingStrategy.Fast.getId()));
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:2904]: Does this work correctly if `sunEnabled` is true and `sunSamplingStrategy` is not present, ie. existing scenes with sun?

**Review by leMaik — APPROVED** [review]

**Comment by jackjt8** [comment]
> Nice! :)
> 
> > Fix issue where enabling draw sun but not enabling sunlight would draw a sun with no contribution to the scene.
> 
> I always assumed this would be a feature so that one can use sky light and emitters only but still have a visual sun?

Correct. It is often used to replace the sun in a skymap.

**Comment by jackjt8** [comment]
Compared to `2.5.0-SNAPSHOT.124.ge9d7e123` (obtained today)

Master/current nightly - 441,874 SPS | 46 s

This PR:
- SSS:OFF/NON_LUMINOUS ~~632,919 SPS | 32 s
- SSS:FAST 438,496 SPS | 46 s (approx. matching master)
- SSS:HIGH_QUALITY 497,111 SPS | 41 s

---

I do not see a performance regression from this change.

**Pull request merged**